code/modules/clothing/clothing.dm

```diff
@@ -198,6 +200,29 @@ BLIND     // can't see anything
 		var/mob/M = loc
 		M.update_inv_shoes()
 
+/obj/item/clothing/shoes/attackby(obj/item/I, mob/user, params)
+	..()
+	if(!can_hold_knives)
+		return
+	if(held_knife)
+		user << "<span class='notice'>There's already something in [src].</span>"
+		return
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/pen) || istype(I, /obj/item/weapon/switchblade))//can hold both regular pens and energy daggers. made for your every-day tactical librarians/murderers.

```

Comment:
Make this a shoe var
`var/list/can_be_hidden = list(obj/item/weapon/kitchen/knife, /obj/item/weapon/pen, /obj/item/weapon/switchblade)`
Then replace this line with
`if(is_type_in_list(I, can_be_hidden))`
That way the hidden list is more flexible and can be overriden by shoe children/creative admins.